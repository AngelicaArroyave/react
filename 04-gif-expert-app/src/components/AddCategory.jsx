import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('')
    
    const onInputChange = (event) => setInputValue(event.target.value)
    const onSubmit = (event) => {
        // Evita que se pierda la informacion
        event.preventDefault()

        const clearValue = inputValue.trim()

        if(clearValue <= 1) return

        onNewCategory(clearValue)
        setInputValue('')
    }

    return (
        <>
            <form onSubmit={ onSubmit } aria-label='form'>
                <input type="text" placeholder="Search Gifs" value={ inputValue } onChange={ event => onInputChange(event) } />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}