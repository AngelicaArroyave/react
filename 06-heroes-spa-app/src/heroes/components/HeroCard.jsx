import { Link } from 'react-router'

const CharactersByHero = ({ alter_ego, characters }) => {
    if(alter_ego === characters) return (<></>)

    return (
        <>
            {
                characters.split(', ').map(character => (
                    <div className="badge badge-outline">{ character }</div>
                ))
            }
        </>
    )
}

export const HeroCard = ({ hero }) => {
    const heroImageUrl = `/public/assets/${ hero.id }.jpg`

    return (
        <>
            <div className="card bg-base-100 w-60 shadow-sm mt-4 animate__animated animate__fadeIn">
                <figure>
                    <img
                        src={ heroImageUrl }
                        alt={ hero.superhero } />
                </figure>
                <div className="card-body">
                    <h2 className="card-title">{ hero.superhero }</h2>
                    <p>{ hero.alter_ego }</p>
                    <p className="text-gray-500">{ hero.first_appearance }</p>
                    <div className="card-actions justify-end">
                        {
                            <CharactersByHero alter_ego={ hero.alter_ego } characters={ hero.characters } />
                        }
                    </div>
                    <Link to={`/hero/${ hero.id }`}>
                        <button className="btn btn-primary btn-block">Show more</button>
                    </Link>
                </div>
            </div>
        </>
    )
}
