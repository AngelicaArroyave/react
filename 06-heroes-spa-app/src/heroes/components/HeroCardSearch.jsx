import { Link } from 'react-router'

export const HeroCardSearch = ({ hero }) => {
    const heroImageUrl = `/public/assets/${hero.id}.jpg`

    return (
        <>
            <div className="flex flex-col items-center justify-center mt-10 mr-4 ml-4">
                <div className="card card-side bg-base-100 shadow-sm w-full lg:w-1/2 items-center">
                    <figure>
                        <img src={heroImageUrl} alt={hero.superhero} className="img-thumbnail w-50 animate__animated animate__fadeInUp" />
                    </figure>
                    <div className="card-body">
                        <h3 className="card-title text-4xl">{hero.superhero}</h3>

                        <p>{hero.alter_ego}</p>
                        <p className="text-gray-500">{hero.first_appearance}</p>

                        <Link to={`/hero/${hero.id}`}>
                            <button className="btn btn-primary btn-block">Show more</button>
                        </Link>
                    </div>
                </div>
            </div>
        </>
    )
}
