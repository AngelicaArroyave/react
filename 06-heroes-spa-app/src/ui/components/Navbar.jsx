import { Link, NavLink, useNavigate } from 'react-router'

export const Navbar = () => {
    const navigate = useNavigate()
    const onLogout = () => {
        navigate('/login', { replace: true })
    }

    return (
        <>
            <div className="navbar bg-neutral text-neutral-content shadow-sm">
                <div className="navbar-start">
                    <div className="dropdown">
                        <div tabindex="0" role="button" className="btn btn-ghost lg:hidden">
                            <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor"> <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M4 6h16M4 12h8m-8 6h16" /> </svg>
                        </div>
                        <ul tabindex="0"
                            className="menu menu-sm dropdown-content bg-neutral-400 rounded-box z-1 mt-3 w-52 p-2 shadow text-gray">
                            <li>
                                <NavLink to="/marvel">Marvel</NavLink>
                            </li>
                            <li>
                                <NavLink to="/dc">DC</NavLink>
                            </li>
                        </ul>
                    </div>
                    <Link className="btn btn-ghost text-xl" to="/">Associations</Link>
                </div>
                <div className="navbar-center hidden lg:flex">
                    <ul className="menu menu-horizontal px-1">
                        <li>
                            <NavLink className={({ isActive }) => isActive ? "bg-info text-black" : ""} to="/marvel">Marvel</NavLink>
                        </li>
                        <li>
                            <NavLink className={({ isActive }) => isActive ? "bg-info text-black" : ""} to="/dc">DC</NavLink>
                        </li>
                    </ul>
                </div>
                <div className="navbar-end">
                    <span className="text-cyan-400 mr-3">Angie</span>
                    <div className="mr-5 hover:bg-info rounded-3xl w-10 h-10 flex items-center justify-center cursor-pointer" onClick={ onLogout }>
                        <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24"><path fill="none" stroke="currentColor" d="M24 12H8m10.5-5c0 .577.665 1.562 1.228 2.294a7.5 7.5 0 0 0 1.745 1.662C22.2 11.445 23.2 12 23.99 12c-.79 0-1.79.556-2.517 1.044a7.5 7.5 0 0 0-1.745 1.662c-.563.732-1.228 1.717-1.228 2.294m-4-10V2.5h-.329A46 46 0 0 1 1.103.605L.75.5H.5v23h.25l.353-.105A46 46 0 0 1 14.171 21.5h.329V17" stroke-width="1"/></svg>
                    </div>
                </div>
            </div>
        </>
    )
}
