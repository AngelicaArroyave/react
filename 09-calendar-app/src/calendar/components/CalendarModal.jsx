import { addHours } from 'date-fns'
import { useState } from 'react'
import DatePicker, { registerLocale } from 'react-datepicker'
import Modal from 'react-modal'
import 'react-datepicker/dist/react-datepicker.css'
import es from 'date-fns/locale/es'

registerLocale('es', es)

const customStyles = {
    content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
    }
}

Modal.setAppElement('#root')

export const CalendarModal = () => {
    const [isOpen, setIsOpen] = useState(true)
    const [formValues, setFormValues] = useState({
        title: 'John',
        notes: 'Dae',
        start: new Date(),
        end: addHours(new Date(), 2)
    })
    const [startDate, setStartDate] = useState(new Date())

    const onInputChanged = ({ target }) => {
        setFormValues({
            ...formValues,
            [target.name]: target.value
        })
    }

    const onDateChanged = (event, changing) => {
        setFormValues({
            ...formValues,
            [changing]: event
        })
    }

    const onCloseModal = () => {
        setIsOpen(false)
    }

    return (
        <Modal className="modal" isOpen={isOpen} style={customStyles} onRequestClose={onCloseModal} overlayClassName="modal-background" closeTimeoutMS={200}>
            <h1>Nuevo Evento</h1>
            <hr />

            <form className="container">
                <div className="form-group mb-2 customDatePickerWidth">
                    <label>Fecha y hora de inicio</label>
                    <DatePicker className='form-control' dateFormat="Pp" showTimeSelect locale="es" timeCaption="Hora"
                    selected={formValues.start} onChange={event => onDateChanged(event, 'start')} />
                </div>

                <div className="form-group mb-2 customDatePickerWidth">
                    <label>Fecha y hora fin</label>
                    <DatePicker className='form-control' dateFormat="Pp" showTimeSelect locale="es" timeCaption="Hora"
                    minDate={formValues.start} selected={formValues.end}
                    onChange={event => onDateChanged(event, 'end')} />
                </div>
                
                <hr />

                <div className="form-group mb-2">
                    <label>Título y notas</label>
                    <input type="text" className="form-control" placeholder="Title and notes" name="title"
                            autoComplete="off" value={formValues.title} onChange={onInputChanged} />
                    <small id="emailHelp" className="form-text text-muted">Descripción corta</small>
                </div>

                <div className="form-group mb-2">
                    <textarea type="text" className="form-control" placeholder="Notes" rows="5" name="notes"
                                value={formValues.notes} onChange={onInputChanged}></textarea>
                    <small id="emailHelp" className="form-text text-muted">Información adicional</small>
                </div>

                <button type="submit" className="btn btn-outline-primary btn-block">
                    <i className="far fa-save"></i>
                    &nbsp;
                    <span>Guardar</span>
                </button>
            </form>
        </Modal>
    )
}
