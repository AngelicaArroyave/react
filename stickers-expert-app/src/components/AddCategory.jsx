import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('')

    const onInputChange = event => {
        setInputValue(event.target.value)
    }

    const onSubmit = event => {
        event.preventDefault()
        const clearInput = inputValue.trim()

        if(clearInput.length <= 1) return

        onNewCategory(clearInput)
        setInputValue('')
    }

    return (
        <>
            <form className='flex flex-col items-center mb-6' onSubmit={ onSubmit }>
                <label className="input">
                    <svg className="h-[1.5em] opacity-50" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
                        <g
                            stroke-linejoin="round"
                            stroke-linecap="round"
                            stroke-width="2.5"
                            fill="none"
                            stroke="currentColor"
                        >
                            <circle cx="11" cy="11" r="8"></circle>
                            <path d="m21 21-4.3-4.3"></path>
                        </g>
                    </svg>
                    <input type="text" required placeholder="Search stickers" value={inputValue} onChange={event => onInputChange(event)} />
                </label>
            </form>
        </>
    )
}