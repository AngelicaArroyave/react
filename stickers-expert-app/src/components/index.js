export * from './AddCategory'
export * from './SkeletonCard'
export * from './StickerGrid'
export * from './StickerItem'