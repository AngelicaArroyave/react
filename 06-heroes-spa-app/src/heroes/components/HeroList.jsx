import { getHeroesByPublisher } from '../helpers'
import { HeroCard } from './HeroCard'

export const HeroList = ({ publisher }) => {
    const heroes = getHeroesByPublisher(publisher)

    return (
        <>
            <div className="grid sm:grid-cols-2 lg:grid-cols-3 gap-4 justify-items-center">
                {
                    heroes.map(hero => (
                        <HeroCard key={ hero.id } hero={ hero } />
                    ))
                }
            </div>
        </>
    )
}
