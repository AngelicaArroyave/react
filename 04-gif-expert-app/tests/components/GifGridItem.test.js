import { render, screen } from '@testing-library/react'
import { GifGridItem } from '../../src/components/GifGridItem'

describe('Test in GifGridItem component', () => {
    const title = 'Rick and Morty'
    const url = 'https://media.giphy.com/media/l0HlE0b7l3j3q/giphy.gif'

    test('Should make match with snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />)

        expect(container).toMatchSnapshot()
    })

    test('Should display the image with the URL and ALT indicated', () => {
        render(<GifGridItem title={title} url={url} />)
        
        const { src, alt } = screen.getByRole('img')

        // Forma 1
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        
        // Forma 2
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Should display the title in the component', () => {
        render(<GifGridItem title={title} url={url} />)

        expect(screen.getByText(title)).toBeTruthy()
    })
})