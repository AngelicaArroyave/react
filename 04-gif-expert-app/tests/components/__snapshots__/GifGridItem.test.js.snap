// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGridItem component Should make match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Rick and Morty"
      src="https://media.giphy.com/media/l0HlE0b7l3j3q/giphy.gif"
    />
    <p>
       
      Rick and Morty
       
    </p>
  </div>
</div>
`;
